rewrite-java/src/main/java/org/openrewrite/java/JavaParser.java

```diff
@@ -162,6 +162,17 @@ static List<Path> dependenciesFromResources(ExecutionContext ctx, String... arti
             version = 8;
         }
 
+        if (version >= 25) {
+            try {
+                return (JavaParser.Builder<? extends JavaParser, ?>) Class
+                        .forName("org.openrewrite.java.Java25Parser")
+                        .getDeclaredMethod("builder")
+                        .invoke(null);
+            } catch (Exception e) {
+                //Fall through, look for a parser on an older version.
+            }
+        }
+
         if (version >= 21) {
```

Comment:
I wonder here we're doing these comparisons wrong; shouldn't we use something like
```suggestion
        if (version > 21) {
            try {
                return (JavaParser.Builder<? extends JavaParser, ?>) Class
                        .forName("org.openrewrite.java.Java25Parser")
                        .getDeclaredMethod("builder")
                        .invoke(null);
            } catch (Exception e) {
                //Fall through, look for a parser on an older version.
            }
        }

        if (version > 17) {
```
And so on... 

That way folks get the next LTS whenever available as that's more likely to fold in parser fixes than a previous version.

Reply:
Then for 26+ Shannon recently proposed adding a `rewrite-java-next` if that requires changes, as an on going effort to support the then-latest-not-LTS version going forward.